### DISCUSSION TITLE
Right-click with press and hold does not work on taskbar, start menu, etc.

### DISCUSSION THREAD

**Discussion by teralomaniac** [LizardByte/community#263] [Sunshine]
Right-click with press and hold in direct touch mode do not seem to work well on win10/11. 

I used the moonlight client for Android and windows, connected two computers with win10 and win11 respectively in direct touch mode, in the taskbar, start menu and other places are unable to right-click with press and hold. But touchpad mode worked well.

Right-click with press and hold worked when using the NVidia Shield GameStream instead of Sunshine, so I’m not sure if it’s a setting I’ve neglected.

**Comment by xmcp** [comment]
I also experience this issue, and I believe it is due to [a hack in Sunshine's code](https://github.com/LizardByte/Sunshine/blob/31e8b798dabf47d5847a1b485f57cf850a15fcae/src/input.cpp#L388) that aims to prevent left-clicking on links by mistake.

Originally in GameStream, when long-pressing the screen, it fires below four events in sequence:

1. (When the finger touches the screen) Left button down;
2. (When the finger leavers the screen) Left button up;
3. (Then immediately) Right button down;
4. (Then immediately) Right button up.

Therefore GameStream will accidentally left-click a link (at the 2nd event) when long-pressing. Sunshine "fixes" this problem by delaying the 2nd event by 10ms, so that the events are now:

1. (When the finger touches the screen) Left button down;
2. 🔼 (When the finger leavers the screen) Right button down;
3. 🔼 (Then immediately) Right button up;
4. **🔻 (Delayed after 10ms) Left button up.**

While this hack works for some browsers like Chrome, unfortunately Windows does not allow right-click for context menu while the left mouse button is down (You can confirm this by simply trying the above 4 steps in sequence).

**Reply by xmcp** [reply]
Reported as an issue at https://github.com/LizardByte/Sunshine/issues/1432
